Week1 [STEP 3] Ann

johyein1017 opened the pull request:
@Tediousday93 

## 고민했던 점
### 매개변수, 반환값이 있는 함수 만들기
처음에 `drawPeperoBodyHeight`(함수)에서 `drawPeperoBody`(함수)를 불러오기하니까 진행이 안되서 방법을 찾아보다가, 새로운 매개변수(`bodyshape`)를 만들어서 상수를 적용하면 될 것 같더라구요...!
`drawPeperoBody`를 불러오려고했던건, 토핑과 몸통을 그리는 값을 몸통 길이만큼 반복하고 싶어서였습니다!
상수 적용까지 해놓고ㅜㅜ 마지막에 모든 함수를 호출하는 새로운 함수(`makePepero`)에 함수구현부에서 살짝 헷갈렸지만 로웬의 조언으로 제 나름의 방식대로 해결했습니다...!

## 조언을 얻고 싶은 부분
### 반환값
```
func drawPeperoBody(body: String, topping: String) -> String {
  return topping + body + topping
}
```
에서 `return`하고 매개변수 이름들을 나열해줬는데요! 
`return \(topping) + \(body) + \(topping)`과 `return topping + body + topping`이 무슨 차이인지 잘 모르겠어요😭😭

### 함수,매개변수 이름 및 함수 구현법(?)
- 알려주신 가이드를 참고하여 진행했는데, 제가 잘 작성하고 있는건지,,,, 조언 부탁드릴게요🙏🏻
- 제가 생각했던 빼빼로 만들기는 원하는 모양대로 몸통과 토핑을 만들고 그걸 원하는 길이만큼 그려내는 함수였고, 마지막 함수를 호출하는 부분에서 원하는 몸통,토핑,길이 값을 입력하면 출력되는 것이었는데.... 생각대로 진행되긴했지만 매개변수 이름도 그렇고 마지막에 모든 함수를 한번에 호출할 수 있게 만들어주는 함수(`makePepero`)가 너무 복잡하게 구현된 것 같아서 더 깔끔하게? 하는 방법은 어떤 것이 있는지 조언 주시면! 힌트대로 다시 한 번 연습해보겠습니다😂
- (속마음 : 매개변수나 반환값을 넣어서 구현하는 함수는 아직도 너무,,,, 어려워요,,,,,,,,,,,,,😭)

Commit:
빼빼로 만들기 함수 추가(xcode 파일)

Commit:
순서도 파일 업로드

file : week1-step3-빼빼로 만들기 순서도.png

Tediousday93 reviewed (changes requested):
안녕하세요 앤님,
이번 스텝도 정말 잘 수행해주셨네요!
고생하셨습니다👍 

아래 코드에서도 코멘트를 확인해주시고, 수정이 완료되시면 디스코드로 머지 의사를 알려주세요 ☺️

---

### 수행한 점

* [x] 정확한 브랜치에 PR 보내기
* [x] PR의 제목/내용을 양식에 맞게 보내기
* [x] 학습 도우미 태그하기

---

* [x] Step2와 같은 프로젝트에서 새로운 파일을 작성하기
* [x] 요구사항에 맞는 함수 만들기
* [x] 다양한 빼빼로 출력하기
* [x] 순서도 만들기

---

</br>

### ✅ 고민했던 점

함수의 파라미터 활용에 대해 많이 고민하셨군요 👍
`drawPeperoBodyHeight` 함수 잘 구현해주셨습니다!
파라미터를 통해 출력할 문자열을 전달받음으로서 함수의 기능이 잘 분리되었어요 👏👏👏

`makePepero` 함수도 잘 구현해주셨습니다! 👍
다만, 전체적으로 파라미터에는 camel case가 지켜지지 않고 있네요.
이 부분은 간단히 고쳐주시면 되겠습니다😊
</br>

### ✅ 해결이 되지 않은 점

* **반환값**
문자열 보간법과 문자열 사이의 연산을 통한 표현의 차이라고 생각됩니다.
문자열 보간법의 사용 방법이 살짝 잘못되어있네요...!
문자열 보간법은 `"\(값)"`의 형태로 사용해야합니다 👀

* **네이밍 / 함수 구현방법**
    * 이름 아주 잘 지어주시고 있어요👍 </br> 아직 어색해서 그렇게 느껴지실 수도 있습니다. 천천히 연습하면 되니 걱정마세요.☺️
    * 전혀 복잡하지 않다고 생각합니다. 여러개의 함수로 나눠 정의해주셔서 기능이 잘 분리되었어요! </br> 더 깔끔하게 작성하고 싶은 욕심이 생기신다면, 정보를 출력하는 기능도 별도의 함수로 분리해볼 수 있답니다~
    * (매개변수/반환값을 갖는 함수는 자주 사용하시다 보면 차차 익숙해지실 거라고 생각해요. 어려운 부분이 생기신다면 언제든 질문 해주세요! 💪)
    * 질문하실 때는 캠핑 사이트 필독사항의 `⁉️ 질문과 답변, 그리고 공유하기`를 참고하면 서로에게 도움이 된답니다!

Commit:
빼빼로 만들기 파일 수정(xcode)

file : CodeStarterCamp_Week1.xcodeproj

Commit:
순서도 수정

file delete : week-step3-빼빼로 만들ᄀ.png
file add : week-step3-빼빼로 만들기 순.png

Commit:
빼빼로 만들기 파일 재수정(xcode)

file : CodeStarterCamp_Week1.xcodeproj

Commit:
순서도 재수정

delete file : week-step3-빼빼 만들기 순서도(수정).png
add file : week-step3-빼빼로 만들기 순서도(재수정).png
